app/src/app/api/v0/auth/verify/route.ts

```diff
@@ -0,0 +1,72 @@
+import { db } from "@/models";
+import { apiHandler } from "@/util/api";
+import { passwordRegex, resetPasswordRequest } from "@/util/validation";
+import bcrypt from "bcrypt";
+import createHttpError from "http-errors";
+import jwt from "jsonwebtoken";
+import { Session } from "next-auth";
+import { NextRequest, NextResponse } from "next/server";
+import { z } from "zod";
+
+const requestVerification = z.object({
+  password: z.string().min(4).regex(passwordRegex),
+  token: z.string(),
+});
+
+export const GET = apiHandler(
+  async (
+    _req: NextRequest,
+    context: { session?: Session; params: Record<string, string> },
+  ) => {
+    const email = context.session?.user.email;
+    const user = await db.models.User.findOne({ where: { email } });
+
+    if (!user) {
+      throw createHttpError.NotFound("User not found!");
+    }
+
+    if (!process.env.RESET_TOKEN_SECRET) {
```

Comment:
Add this secret (with placeholder value) to:
- env.example
- github action (tests)